### DISCUSSION TITLE
Autotilt current mesh probe points

### DISCUSSION THREAD

**Discussion by gothcha** [Jyers/Marlin#398] [Q&A] [answered]
@Jyers When the G29 J command tilts the mesh it uses 3 points, which can be user defined, however when the 'Autotilt current mesh' from the Level menu is used it probes close to the four corners of the bed and in my case bumps against the Ymax end stop. 

Is this behaviour hard coded? and why does it not use the same 3 point system and coordinates?

I wish to thank you and all the great code writers who have contributed to this fabulous firmware, I love the new additions, menus and icons.

**Comment by kted** [comment, marked as answer]
Try G29 J2 for 4-point tilt check
From the documentation:
`J<int>`  If a value is provided, probe a grid with the given number of points, squared

**Reply by gothcha** [reply]
I think I have found the setting
In creality_dwin.cpp 

#if ENABLED(AUTO_BED_LEVELING_UBL)
  struct UBL_Settings {
    bool viewer_asymmetric_range = false;
    bool viewer_print_value = false;
    uint8_t tilt_grid = 2;

with tilt_grid set to >1 it runs G29 J2 otherwise it runs G29 J, so if this is set to 1 it should run the standard 3point tilt.

I will try modifying this parameter and check it works. This may be a better default?

**Reply by Jyers** [reply]
This value can be modified from the screen in the UBL Settings menu. However, the most recent fixes should fix the problem of the printhead hitting the endstop.

**Reply by Jyers** [reply]
Just saw your post in #189. I would not recommend setting the `PROBING_MARGIN` to 0, if you want to probe the whole mesh it should at least be set to the same value as `MESH_INSET`

**Reply by gothcha** [reply]
Thanks Jacob, point well taken, I think I was initially a little unclear if the Probing_Margins were from bed size or mesh size so I used 0 as a test. I believe that they are from the bed edge to the probe position? So I will se the individual offsets on each side the same as the Mesh_Insets.

On the tilt, I did earlier see the menu item but it wasn't obvious what the tilt mesh size did, or that 1 is 3 points and 2 is 4 points..., I guess there is no reason to have anything other than 3/4 points is there? Could the menu give the option for number of points, this would I think make it easier to use. It must be hard to keep the WiKi up to date with so many great releases!

**Reply by Jyers** [reply]
The reason someone might want a larger tilt mesh is that if your bed is quite warped a 3/4 point mesh might not actually give you a good tilt. At least that's the reasoning I suspect. I think I'll keep the tilt size setting in the same area as it is something most people will set and leave. It can be saved to eeprom so you don't need to reset it every time.

**Reply by gothcha** [reply]
I just checked and the tilt size does not appear to be persistent, it resets to the defined value of 2 on power-up.    

uint8_t tilt_grid = 2;

**Reply by Jyers** [reply]
Hmmm, let me see if I can add that to the eeprom settings
